proposed/2023/NuGetAudit-without-nuget.org-as-packageSource.md

```diff
@@ -0,0 +1,190 @@
+
+While the design of NuGetAudit attempts to make it easy for upstreaming package sources to also upstream the `VulnerabilityInfo` resource, the NuGet team and customers are reliant of the feed implementor to implement the change.
+An upstreaming package source is one that either aggregates multiple package sources or caches packages from another package source.
+Upstreaming isn't a concept that exists in NuGet, so servers have to present to clients as if all data (packages, search results, vulnerability information) originates from itself.
+
+However, there are also cases where customers are not using upstreaming feeds, but instead curate approved 3rd party packages and manually upload them to an internal feed.
+There are also other scenarios where customers do not use nuget.org (directly) as a package source.
+
+It's possible for customers to use [Package Source Mapping](https://learn.microsoft.com/nuget/consume-packages/package-source-mapping) to add nuget.org as a package source, but configured in a way that NuGet will never restore packages from this package source.
+However, some developers might consider this an unacceptable risk and prefer another way to get a VDB that cannot accidentally be used to get packages.
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+*NuGet.Config* files have a new section `<auditSources>` where the URL(s) of any audit sources can be specified, which will be used only to download a VDB, and will not be used for downloading packages.
+Like `<packageSources>`, `<auditSources>` can contain zero, one, or more sources, and supports `<clear />`.
+When there is at least one source in `auditSources`, NuGet should no longer use `packageSources` to get a VDB.
+
+For example, consider a developer at Contoso using a company-internal source as a single package source, and using nuget.org as an audit source:
+
+```xml
+<configuration>
+  <packageSources>
+    <clear />
+    <add key="contoso" value="https://internal.corp.contoso.com/packages/nuget/index.json" >
+  </packageSources>
+  <auditSources>
+    <clear />
+    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" >
+  </auditSources>
+</configuration>
+```
+
+Any URL specified in the `auditSources` section must implement the [NuGet Server API V3 protocol](https://learn.microsoft.com/nuget/api/overview).
+However, the server will only need to implement the [`VulnerabilityInfo` resource](https://learn.microsoft.com/en-us/nuget/api/vulnerability-info) (and service index), and can avoid implementing all the other resources that are required to be used as a package source.
+
```

Comment:
[Non MVP, could wait for customer feedback] What's the format for popular audit sources? Could this support some generic format as well like say a csv file?

Reply:
It's officially documented on our docs site: https://learn.microsoft.com/en-us/nuget/api/vulnerability-info

It was proposed in the original NuGetAudit feature: https://github.com/NuGet/Home/blob/dev/proposed/2022/vulnerabilities-in-restore.md#dedicated-vulnerability-information-resource

I'm resolving this conversation as out of scope, since this spec is about a specific NuGetAudit feature, not a broad NuGetAudit discussion.

> Could this support some generic format as well like say a csv file?

This is already addressed at the end of the spec.